app/index/get.controller.js

```diff
@@ -0,0 +1,11 @@
+'use strict'
+
+// Local dependencies
+const countries = require('../../common/middleware/countries')
+
+module.exports = (req, res) => {
```

Comment:
I would use another name, since _get controller_ is not telling anything about what is it about

Reply:
I am not sure I am a fan of get/post naming.

I’d prefer it say what it does, index isn’t very helpful either.

maybe something like:

* `app/setup/get-details` / `app/setup/post-details` (I am not sure if controller is even helpful. Aren't all `.js` (that aren't `.test.js`) files going to be controllers?)
* `app/one-off/show-form` / `app/one-off/submit-form`
* `app/dd-form/show` / `app/dd-confirm/show`

Reply:
In fact, in my opinion, a modular approach would mean each directory contains a single controller. I don’t really see the merit in splitting the `post` from the `get` as for the user, its one-in-the-same, if they can load a form, you should be able to split a form.

I’d prefer a structure like
```
├── app
|   ├──submit-details
|       ├── index.js // all the js that makes it work post/get/routes
|       ├── index.test.js // tests for functionality
|       ├── index.njk // the template for this
|       ├── style.scss // could even drop all the styles here too
```

Reply:
@rauligs Thanks for the comments, initially we named the controllers to have the feature name as well, but later we had a discussion that the actual parent folder is named the same, so that is why we decided to have a structure like this:
```
- /feature-one/
--- get.controller.js
--- get.controller.it.tests.js
--- post.controller.js
--- post.controller.it.tests.js
--- delete.controller.js
--- delete.controller.it.tests.js
--- put.controller.js
--- put.controller.it.tests.js
```
as @jonheslop described as well

@jonheslop @rauligs I think the name `setup` is much better for submitting the details, but I was not sure should we have dedicated `index` feature folder for the root of the project (which is currently `setup`) to be explicit what is the `/` home feature.

@jonheslop @rauligs @maxcbc let me know if you want me to change something

Thanks for the review!

Reply:
@DanailMinchev Re `/index/` for the project root. I am trying to think what that would do in reality, you can't just land on `direct-debit.payments.service.gov.uk` it would just error, does that make it confusing to have one? Maybe it should just redirect straight to our `/error/` page?

Reply:
@jonheslop good point, yes, I think if we are not having the `/` page we should error, I will need to clarify that: what is the initial page.
For the `/error` page I think we need to add error middleware + page (including NAXSI errors) which is part of the other story I think

Reply:
Basically I'm of the view that each controller should have it's own file. This makes it much clearer in my view where functionality lives, how each file is named is debatable, the http verb names to make sense but as do more descriptive names, as there may be occasions where you use the same controller to handle more than one http verb.

Another reason for this is that each controller should have it's own individual test file. Test files get mahussive even when they only test one controller, a explicit mapping between `<filename>.js` and `<filename>.test.js` makes this absolutely clear. This is important for controllers and doubly so for utilities/clients e.t.c as the vast majority of code should end up in the latter.

@DanailMinchev I'm not sure there is any need for the `.it` in the filename. We only seem test each thing one way so I think just `<filename>.test.js` is fine. For example:
- controllers -> we only test via supertest
- utilities -> we only test as unit tests
- clients -> we only test via Pact

Reply:
Thanks @maxcbc addressed in [1546b0b]